src/web/core/src/main/java/org/geoserver/web/data/store/DefaultDataStoreEditPanel.java

```diff
@@ -160,13 +160,15 @@ private Panel getInputComponent(final String componentId, final IModel paramsMod
         final Class<?> binding = paramMetadata.getBinding();
         final List<Serializable> options = paramMetadata.getOptions();
 
+        boolean visible = true;
+
         Panel parameterPanel;
         if("dbtype".equals(paramName) || "filetype".equals(paramName)) {
```

Comment:
Given that adding that flag to the parameters in GeoTools could affect the behaviour of arbitrary downstream applications, that would count as an API change, correct?